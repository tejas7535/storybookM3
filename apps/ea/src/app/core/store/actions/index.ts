export * from './calculation-parameters';
export * from './calculation-result';
export * from './product-selection';
export * from './settings';
export * from './storage-messages';
