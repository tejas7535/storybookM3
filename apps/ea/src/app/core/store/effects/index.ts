export * from './calculation-parameters';
export * from './calculation-result';
