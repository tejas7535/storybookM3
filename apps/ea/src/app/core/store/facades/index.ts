export * from './calculation-parameters/calculation-parameters.facade';
export * from './settings/settings.facade';
