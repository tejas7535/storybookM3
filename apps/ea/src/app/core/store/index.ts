export * from './facades';
export * from './reducers';
export * from './selectors';
export * from './store.module';
