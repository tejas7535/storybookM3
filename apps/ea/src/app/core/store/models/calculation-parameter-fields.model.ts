export type CalculationParameterField =
  | 'rpm'
  | 'load'
  | 'time'
  | 'lubrication'
  | 'contamination'
  | 'operatingTemperature'
  | 'ambientTemperature'
  | 'energySource'
  | 'rotatingCondition';
