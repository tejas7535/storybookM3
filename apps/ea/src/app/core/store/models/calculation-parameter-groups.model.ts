export type CalculationParameterGroup =
  | 'load'
  | 'lubrication'
  | 'contamination'
  | 'operatingTemperature'
  | 'ambientTemperature'
  | 'time'
  | 'energySource'
  | 'rotatingCondition'
  | 'externalHeatflow'
  | 'conditionOfRotation'
  | 'operatingTimeAndTemperature';
