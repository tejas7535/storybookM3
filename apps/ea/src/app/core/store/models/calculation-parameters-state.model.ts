import {
  CatalogServiceLoadCaseData,
  CatalogServiceOperatingConditions,
} from '@ea/core/services/catalog.service.interface';
import { Greases } from '@ea/shared/constants/greases';

export interface CalculationParametersState {
  operationConditions: CalculationParametersOperationConditions;
  calculationTypes: CalculationParametersCalculationTypes;
  isInputInvalid: boolean;
}

export interface CalculationParametersOperationConditions {
  operatingTemperature: number;
  ambientTemperature: number;

  load: {
    axialLoad: number;
    radialLoad: number;
  };
  rotation: {
    typeOfMotion: CatalogServiceLoadCaseData['IDSLC_TYPE_OF_MOVEMENT'];
    rotationalSpeed: number;
    shiftFrequency: number;
    shiftAngle: number;
  };

  contamination: CatalogServiceOperatingConditions['IDL_CLEANESS_VALUE'];

  lubrication: {
    lubricationSelection: 'grease' | 'oilBath' | 'oilMist' | 'recirculatingOil';
    grease: {
      selection: 'typeOfGrease' | 'isoVgClass' | 'viscosity';
      typeOfGrease: {
        typeOfGrease: typeof Greases[number]['value'];
      };
      isoVgClass: {
        isoVgClass: number;
      };
      viscosity: {
        ny40: number;
        ny100: number;
      };
    };
    oilBath: {
      selection: 'isoVgClass' | 'viscosity';
      isoVgClass: {
        isoVgClass: number;
      };
      viscosity: {
        ny40: number;
        ny100: number;
      };
    };
    oilMist: {
      selection: 'isoVgClass' | 'viscosity';
      isoVgClass: {
        isoVgClass: number;
      };
      viscosity: {
        ny40: number;
        ny100: number;
      };
    };
    recirculatingOil: {
      selection: 'isoVgClass' | 'viscosity';
      isoVgClass: {
        isoVgClass: number;
      };
      viscosity: {
        ny40: number;
        ny100: number;
      };
    };
  };

  externalHeatFlow: number;

  conditionOfRotation: 'innerring' | 'outerring';
}

export type CalculationType =
  | 'frictionalPowerloss'
  | 'ratingLife'
  | 'lubrication'
  | 'emission'
  | 'overrollingFrequency';

export type CalculationParametersCalculationTypes = Record<
  CalculationType,
  { selected: boolean; visible: boolean; disabled: boolean }
>;

export interface CalculationParametersCalculationTypeConfig {
  name: CalculationType;
  selected: boolean;
  disabled: boolean;
  label: string;
  svgIcon?: string;
  icon?: string;
}
