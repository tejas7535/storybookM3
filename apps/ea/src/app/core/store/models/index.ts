export * from './calculation-parameter-groups.model';
export * from './calculation-parameters-state.model';
export * from './calculation-result-preview-data.model';
export * from './calculation-result-report-input.model';
export * from './calculation-result-state.model';
export * from './catalog-calculation-result-state.model';
export * from './co2-upstream-calculation-result-state.model';
export * from './friction-calculation-result-state.model';
export * from './product-selection-state.model';
export * from './settings-state.model';
