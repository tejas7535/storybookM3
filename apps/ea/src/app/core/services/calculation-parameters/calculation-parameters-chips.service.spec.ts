import { CalculationParametersEnergySource } from '@ea/core/store/models';
import { TranslocoService } from '@jsverse/transloco';
import { createServiceFactory, SpectatorService } from '@ngneat/spectator/jest';

import { provideTranslocoTestingModule } from '@schaeffler/transloco/testing';

import { DownstreamOperatingConditions } from '../downstream-calculation.service.interface';
import { CalculationParametersChipsService } from './calculation-parameters-chips.service';
import {
  CalculationChip,
  ElectricityRegionOption,
  FossilOriginOption,
} from './calculation-parameters-form.interface';

describe('CalculationParametersChipsService', () => {
  let spectator: SpectatorService<CalculationParametersChipsService>;

  const createService = createServiceFactory({
    service: CalculationParametersChipsService,
    imports: [provideTranslocoTestingModule({ en: {} })],
    providers: [
      {
        provide: TranslocoService,
        useValue: {
          translate: jest.fn((translateKey) => translateKey),
        },
      },
    ],
  });

  beforeEach(() => {
    spectator = createService();
  });

  describe('getContaminationChip', () => {
    it('should return a CalculationChip with correct values', () => {
      const contaminationOptions = [
        { value: 'LB_HIGH_CLEANLINESS', label: 'High Cleanliness' },
        { value: 'LB_STANDARD_CLEANLINESS', label: 'Standard Cleanliness' },
      ];

      const contamination = 'LB_STANDARD_CLEANLINESS';

      const result: CalculationChip = spectator.service.getContaminationChip(
        contaminationOptions,
        contamination
      );

      expect(result).toMatchSnapshot();
    });
  });

  describe('getEnergySourceChip', () => {
    let type: CalculationParametersEnergySource['type'];
    let fossilOrigin:
      | DownstreamOperatingConditions['fossilEmissionFactor']
      | undefined;

    let electricityRegion:
      | DownstreamOperatingConditions['electricEmissionFactor']
      | undefined;

    const fossilOptions: FossilOriginOption[] = [
      { value: 'LB_GASOLINE_E10', label: 'Gasoline E10' },
    ];
    const electricityRegionOptions: ElectricityRegionOption[] = [
      { value: 'LB_AUSTRALIA', label: 'Australia' },
      { value: 'LB_EUROPEAN_UNION', label: 'European Union' },
    ];

    beforeEach(() => {});
    describe('when type is electric', () => {
      beforeEach(() => {
        type = 'electric';
      });
      it('should return a CalculationChip with correct values', () => {
        electricityRegion = 'LB_AUSTRALIA';

        const result: CalculationChip = spectator.service.getEnergySourceChip(
          type,
          fossilOrigin,
          electricityRegion,
          fossilOptions,
          electricityRegionOptions
        );

        expect(result).toMatchSnapshot();
      });

      it('should return correct value for European union', () => {
        electricityRegion = 'LB_EUROPEAN_UNION';

        const result: CalculationChip = spectator.service.getEnergySourceChip(
          type,
          fossilOrigin,
          electricityRegion,
          fossilOptions,
          electricityRegionOptions
        );

        expect(result).toMatchSnapshot();
      });

      it('should return empty for non option value', () => {
        electricityRegion = 'LB_FRANCE';

        const result: CalculationChip = spectator.service.getEnergySourceChip(
          type,
          fossilOrigin,
          electricityRegion,
          fossilOptions,
          electricityRegionOptions
        );

        expect(result).toMatchSnapshot();
      });
    });

    describe('when type is fossil', () => {
      beforeEach(() => {
        type = 'fossil';
      });
      it('should return a CalculationChip with correct values', () => {
        fossilOrigin = 'LB_GASOLINE_E10';

        const result: CalculationChip = spectator.service.getEnergySourceChip(
          type,
          fossilOrigin,
          electricityRegion,
          fossilOptions,
          electricityRegionOptions
        );

        expect(result).toMatchSnapshot();
      });

      it('should return empty value for non option value', () => {
        fossilOrigin = 'LB_CNG';

        const result: CalculationChip = spectator.service.getEnergySourceChip(
          type,
          fossilOrigin,
          electricityRegion,
          fossilOptions,
          electricityRegionOptions
        );

        expect(result).toMatchSnapshot();
      });
    });
  });

  describe('getTimeChip', () => {
    it('should return a CalculationChip with correct values', () => {
      const numberOfHours = 2000;

      const result: CalculationChip =
        spectator.service.getTimeChip(numberOfHours);

      expect(result).toMatchSnapshot();
    });
  });
});
