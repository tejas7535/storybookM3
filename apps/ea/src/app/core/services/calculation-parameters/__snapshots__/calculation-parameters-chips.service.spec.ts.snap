// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculationParametersChipsService getContaminationChip should return a CalculationChip with correct values 1`] = `
Object {
  "icon": "mop",
  "label": "operationConditions.contamination.title",
  "text": "Standard Cleanliness",
}
`;

exports[`CalculationParametersChipsService getEnergySourceChip when type is electric should return a CalculationChip with correct values 1`] = `
Object {
  "icon": "bolt",
  "label": "operationConditions.energySource.title",
  "newTag": true,
  "text": "operationConditions.energySource.electricityChipText",
}
`;

exports[`CalculationParametersChipsService getEnergySourceChip when type is electric should return correct value for European union 1`] = `
Object {
  "icon": "bolt",
  "label": "operationConditions.energySource.title",
  "newTag": true,
  "text": "operationConditions.energySource.electricityChipText",
}
`;

exports[`CalculationParametersChipsService getEnergySourceChip when type is electric should return empty for non option value 1`] = `
Object {
  "icon": "bolt",
  "label": "operationConditions.energySource.title",
  "newTag": true,
  "text": "operationConditions.energySource.electricityChipText",
}
`;

exports[`CalculationParametersChipsService getEnergySourceChip when type is fossil should return a CalculationChip with correct values 1`] = `
Object {
  "icon": "bolt",
  "label": "operationConditions.energySource.title",
  "newTag": true,
  "text": "Gasoline E10",
}
`;

exports[`CalculationParametersChipsService getEnergySourceChip when type is fossil should return empty value for non option value 1`] = `
Object {
  "icon": "bolt",
  "label": "operationConditions.energySource.title",
  "newTag": true,
  "text": "",
}
`;

exports[`CalculationParametersChipsService getTimeChip should return a CalculationChip with correct values 1`] = `
Object {
  "icon": "acute",
  "label": "operationConditions.time.title",
  "newTag": true,
  "text": "2000Â h",
}
`;
